File: rent-now-backend/src/main/java/pl/holowinska/rentnowbackend/model/rq/AccommodationCriteriaRQ.java
package pl.holowinska.rentnowbackend.model.rq;

import lombok.Builder;
import lombok.Data;
import pl.holowinska.rentnowbackend.model.enums.ConvenienceType;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

@Data
@Builder
public class AccommodationCriteriaRQ {

    private LocalDate startDate;
    private LocalDate endDate;
    private String city;
    private String street;
    private BigDecimal squareFootage;
    private BigDecimal minPrice;
    private BigDecimal maxPrice;
    private List<ConvenienceType> conveniences;
}

File: rent-now-ui/src/app/data/model/rq/AccommodationCriteriaRQ.ts
import {ConvenienceType} from "../common/ConvenienceType";

export interface AccommodationCriteriaRQ {

  startDate: Date;
  endDate: Date;
  city?: string;
  street?: string;
  squareFootage?: number;
  minPrice?: number;
  maxPrice?: number;
  conveniences?: ConvenienceType[];
  maxNoOfPeople?: number;
  name?: string;
}
